### ISSUE TITLE
New playbooks not available in job template dropdown after project sync

### ISSUE THREAD

**Issue by packe1bb** [ansible/awx#11635]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

First off, I am completely new to reporting/searching for issues, and as such cannot seem to search for a duplicate (but I tried!) so apologies if this is a duplicate :(

I have noticed behavior where I push new playbooks for new roles to my GitLab project, sync the project in AWX (which is successful), but then when trying to create a new job template for the new role, the playbook is not an option from the dropdown list. However, the top level .yml playbook IS correct syntax (double checked with a YAML linter (which I also have enabled in VSCode). Also, if I create an entirely new GitLab repo and push the exact same code to it and sync with a new project in AWX, the playbooks are available in the dropdown.

### AWX version

19.2.2

### Installation method

docker on linux

### Modifications

no

### Ansible version

2.11.3

### Web browser

Chrome

### Steps to reproduce

Create GitLab repo and a playbook using roles, sync project to repo, create job template from the project and select top level playbook from the dropdown. Create second playbook in same repo, sync project to repo again, and attempt to create a job template for the new playbook using the dropdown for playbooks
![new_playbook](https://user-images.githubusercontent.com/78568157/151602864-daf99922-7be7-4743-bf2f-03c4288e94a8.PNG)
![new_repo](https://user-images.githubusercontent.com/78568157/151602866-f9dfcb68-2429-491b-9fa3-77b6008626ed.PNG)
![new_selectable_playbooks](https://user-images.githubusercontent.com/78568157/151602867-eceb0612-69bc-474c-b536-d5f8229b1232.PNG)
![original_playbook](https://user-images.githubusercontent.com/78568157/151602868-2cdddb0f-7bed-4683-8191-ad21af430e21.PNG)
![original_repo](https://user-images.githubusercontent.com/78568157/151602869-9d774d38-ef42-46c3-9ffa-b031225cc88c.PNG)
![original_selectable_playbooks](https://user-images.githubusercontent.com/78568157/151602870-d42f44e4-d012-4d74-86be-67efad74fe09.PNG)
![projects](https://user-images.githubusercontent.com/78568157/151602871-e5bcdb4b-c7bf-4426-8085-089d25a9155a.PNG)
.

### Expected results

When new playbooks are pushed to GitLab, they should be selectable in new job templates after project sync.

### Actual results

New playbooks are not selectable, unless new repo is made containing the same playbooks and synced to AWX.

### Additional information

Behavior is present in fresh installs of FireFox and Edge browsers as well.

**Comment by nixocio** [comment]
Hello,

We have not heard of any other report of this issue, if other people on the mail listing are seeing this, we can re-open.

If you need help troubleshooting or are looking for help using AWX, try our mailing list or IRC channel:

#ansible-awx on[ https://libera.chat/](https://libera.chat/)
https://groups.google.com/forum/#!forum/awx-project

**Comment by Ganeshk1994** [comment]
Hello @nixocio ,

We are facing the same issue as @packe1bb .  Can you please help on troubleshooting this issue.

**Comment by prtkdave** [comment]
Please reopen. I am facing same issue. 

Even after project sync, latest added playbooks are missing from drop-down.

**Comment by laimis9133** [comment]
Same issue - new GitLab repo playbooks are not visible in the dropdwon menu even after sync. Older playbooks in repo are visible, but new ones created in the same folder do not appear. Please reopen.

**Comment by djjudas21** [comment]
What's going on with this issue? I'm willing to accept user error on this but I've deployed a greenfield AWX installation and I can't get any of my simple playbooks to show up when creating Templates. The Project is syncing successfully and showing the correct commit hash.

**Comment by neduz** [comment]
I'm having same issue, fresh AWX 24.6.1 installed with AWX operator. Getting inventory from a git repo works, changing branch on repo and re-syncing inventory is proven to work. Accessing playbooks that were on the original git checkout works, but after adding a new playbook, there seems to be no way of getting it populated in the dropdown. I've checked the commit-tag, I've checked via the GitLab webui the file is there, but AWX doesn't allow me to list it, and hence will also not let me create new Templates or update existing ones to point to the new playbook yaml.
**UPDATE**: I hadn't actually checked the playbook with a local run of ansible-playbook, I had a formatting error resulting in the yaml file not to be a valid playbook. As soon as I fixed it, re-synced project, I got it in the dropdown.

**Comment by laimis9133** [comment]
One more addition to this issue: a tag in the repository itself might be needed for AWX to pick it up

**Comment by lucasvivian** [comment]
Same issue here. Using AWX version 24.6.1. It does not happens all the time. Some few playbooks are being affected by this bug. I have checked if playbooks are being shown on the project folder after project sync, and it does. I also run ansible-playbook in check mode waiting for any formatting error, but I have none.

UPDATE: I found a solution for my case. It seams AWX does not allow some identations in playbooks. I was setting values for "hosts", "connection" and "gather_facts" parameters and aligning the values in the same column. After remove spaces, playbooks showed up in AWX.

<img width="248" height="92" alt="Image" src="https://github.com/user-attachments/assets/97f07f3f-37e4-403c-96bf-82b6bc9f8df1" />
